src/main/java/com/code_intelligence/jazzer/android/DexFileManager.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.android;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+public class DexFileManager {
+  private final static int MAX_READ_LENGTH = 2000000;
+
+  public static String[] toStringArray(ArrayList<String> list) {
+    String[] result = new String[list.size()];
+
+    for (int i = 0; i < list.size(); i++) {
+      result[i] = list.get(i).toString();
+    }
+
+    return result;
+  }
+
+  public static byte[] getBytecodeFromDex(String jarPath, String dexFile, long offset)
+      throws IOException {
+    try (JarFile jarFile = new JarFile(jarPath)) {
+      Enumeration<JarEntry> allEntries = jarFile.entries();
+
+      while (allEntries.hasMoreElements()) {
+        JarEntry entry = allEntries.nextElement();
+        if (entry.getName().equals(dexFile)) {
+          // Read dex file
+          try (InputStream is = jarFile.getInputStream(entry)) {
+            int bufSize = (int) Math.min(MAX_READ_LENGTH, entry.getSize());
```

Comment:
Wouldn't this logic silently truncate long files? We have a polyfill for `InputStream#readAllBytes` in `src/main/java/com/code_intelligence/jazzer/Jazzer.java` that you could extract and use here.